src/fluree/db/api/decode.cljc

```diff
@@ -0,0 +1,128 @@
+                           (assoc stats :properties (class-properties props ns-codes))
+                           stats)]
+       (assoc acc class-iri decoded-stats)))
+   {}
+   class-stats))
+
+(defn invert-namespace-codes
+  "Inverts namespace-codes map from {code -> ns} to {ns -> code}."
+  [ns-codes]
+  (reduce-kv #(assoc %1 %3 %2) {} ns-codes))
+
+(defn- decode-sid-set
+  "Converts a set of SIDs to a vector of IRIs."
+  [sid-set ns-codes]
+  (when (seq sid-set)
+    (vec (map #(iri/sid->iri % ns-codes) sid-set))))
+
+(defn- merge-property-hierarchy
+  "Merges property hierarchy (subPropertyOf) into property stats.
+   Returns property stats with :sub-property-of added where applicable."
+  [property-stats schema ns-codes]
+  (let [pred-map (get schema :pred {})]
+    (reduce-kv
+     (fn [acc prop-iri prop-stats]
+       (let [;; Find the SID for this property IRI
+             prop-sid (some (fn [[sid prop-data]]
+                              (when (= (:iri prop-data) prop-iri)
+                                sid))
+                            pred-map)
+             ;; Get parent properties (subPropertyOf)
+             parent-props (when prop-sid
+                            (get-in pred-map [prop-sid :parentProps]))
+             ;; Decode parent SIDs to IRIs
+             parent-iris (decode-sid-set parent-props ns-codes)]
+         (if parent-iris
+           (assoc acc prop-iri (assoc prop-stats :sub-property-of parent-iris))
+           (assoc acc prop-iri prop-stats))))
+     {}
+     property-stats)))
+
+(defn- merge-class-hierarchy
+  "Merges class hierarchy (subClassOf) into class stats.
+   Returns class stats with :subclass-of added where applicable."
+  [class-stats schema ns-codes]
+  (let [pred-map (get schema :pred {})]
+    (reduce-kv
+     (fn [acc class-iri class-data]
+       (let [class-sid (some (fn [[sid class-info]]
+                               (when (= (:iri class-info) class-iri)
+                                 sid))
+                             pred-map)
+             parent-classes (when class-sid
+                              (get-in pred-map [class-sid :subclassOf]))
+             parent-iris (decode-sid-set parent-classes ns-codes)]
+         (if parent-iris
+           (assoc acc class-iri (assoc class-data :subclass-of parent-iris))
+           (assoc acc class-iri class-data))))
+     {}
+     class-stats)))
+
+(defn ledger-info
+  "Decodes ledger info by converting SIDs to IRIs and preparing for external consumption.
+   Merges schema hierarchy (subClassOf, subPropertyOf) into stats for classes and properties."
+  [info]
+  (let [ns-codes (:namespace-codes info)
+        schema (:schema info)
+        props (sid-keys (get-in info [:stats :properties]) ns-codes)
+        class-stats (classes (get-in info [:stats :classes] {}) ns-codes)
+        ;; Merge hierarchy info into stats
+        props-with-hierarchy (merge-property-hierarchy props schema ns-codes)
```

Comment:
Seems like this is a no-op for `props`, since they will never have `parent-iris` and so will just add the key and value unchanged back to the map.